Modules/Loadable/VesselEditor/Logic/vtkSlicerVesselEditorLogic.cxx

```diff
@@ -405,4 +405,36 @@ ::SplitTubeInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode, int tubeID, d
       break;
     }
   }
-}
\ No newline at end of file
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerVesselEditorLogic
+::RenumberTubesInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode)
+{
+  if (!spatialNode)
+  {
+    return ;
+  }
+
+  TubeNetType* spatialObject = spatialNode->GetSpatialObject();
+  char childName[] = "Tube";
+  TubeNetType::ChildrenListType* tubeList =
+    spatialObject->GetChildren(spatialObject->GetMaximumDepth(), childName);
```

Comment:
I am using it before the tree creation. So am not considering parent child relationship and just assigning all the tubes to one single parent.